test/unit/test_quilbase.py

```diff
@@ -1004,3 +1022,223 @@ def test_get_qubits(self, swap_phase: SwapPhases, frame_a: Frame, frame_b: Frame
         expected_qubits = set(frame_a.qubits + frame_b.qubits)
         assert swap_phase.get_qubits() == set([q.index for q in expected_qubits if isinstance(q, Qubit)])
         assert swap_phase.get_qubits(False) == expected_qubits
+
+
+@pytest.mark.parametrize(
+    ("left", "right"),
+    [
+        (MemoryReference("ro"), MemoryReference("bar")),
+        (MemoryReference("foo", 5, 10), 5),
+        (MemoryReference("bar"), 3.2),
+    ],
+)
+class TestClassicalMove:
+    @pytest.fixture
+    def move(self, left: MemoryReference, right: Union[MemoryReference, int, float]) -> ClassicalMove:
+        return ClassicalMove(left, right)
+
+    def test_out(self, move: ClassicalMove, snapshot: SnapshotAssertion):
+        assert move.out() == snapshot
```

Comment:
Test `str` as well?

Reply:
Also applies to the rest of the tests

Reply:
I added it to the `SimpleInstruction` tests, but not any of the `AbstractInstruction`s (`ClassicalMove` and anything else besides `Halt`, `Nop`, and `Wait`) as they just fallback to `__repr__` for their string implementation.